test/integration/server.h

```diff
@@ -424,25 +424,49 @@ class IntegrationTestServer : public Logger::Loggable<Logger::Id::testing>,
   void
   waitForCounterEq(const std::string& name, uint64_t value,
                    std::chrono::milliseconds timeout = std::chrono::milliseconds::zero()) override {
-    ASSERT_TRUE(TestUtility::waitForCounterGe(statStore(), name, value, time_system_, timeout));
+    AssertionResult res =
+        TestUtility::waitForCounterGe(statStore(), name, value, time_system_, timeout);
+#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+    ASSERT_TRUE(res);
+#else
+    RELEASE_ASSERT(res, fmt::format("timed out waiting for {} == {}", name, value));
+#endif
```

Comment:
Make this pattern a macro into `fuzz/utility.h`

Reply:
I was trying to do something like 
```
#define ASSERT_ON_FUZZ(x, msg) \
#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION \
ASSERT_TRUE(x); \
#else \
RELEASE_ASSERT(x, msg); \
#endif
```
but it doesn't seem to be possible to do `#ifndef ...` inside a `define` statement.

Asra and I talked about just doing
```
#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION 
#define ASSERT_TRUE(x) RELEASE_ASSERT(x)
#endif 
```
but since the gtest macros are more than just assertions and allow you to stream output to them like `ASSERT_TRUE(x) << "error msg"`, this might lead to some typing issues in some of the tests

Did you have any other suggestions for this?

Reply:
@lizan friendly ping

Reply:
I'm struggling to think of a clever way to redefine `ASSERT_TRUE` to fail because of the streaming message issue. There are some usages of this in Envoy. (I considered injecting an abort if the condition is false, etc). None of our ASSERTs support streaming messages. 

I actually expect most e2e fuzzing tests have this issue (not catching ASSERT's) and it's worth making them fail or having a specific alternative.
I can't tell if it's best to create fuzzer specific methods to have yet another wraper for `TestUtility::waitForCounterEq`. 

Would appreciate suggestions... @htuch maybe?

Reply:
Instead of:
```
#define ASSERT_ON_FUZZ(x, msg) \
#ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION \
ASSERT_TRUE(x); \
#else \
RELEASE_ASSERT(x, msg); \
#endif
```
can you have
```
#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
#define ASSERT_ON_FUZZ(x, msg) RELEASE_ASSERT(x, msg)
#else 
#define ASSERT_ON_FUZZ(x, msg) ASSERT_TRUE(x)
#endif
```
?

In any case, I'm not sure I fully grok why the replacement needs to happen.

Reply:
Yeah that's what I meant. You might be able to just reuse `FUZZ_ASSERT` name.

Reply:
@htuch thanks for that, that makes sense.

The reason behind the replacement is that `ASSERT_TRUE` just exits the function early and logs the error and considers the test a failure, while `RELEASE_ASSERT` actually crashes the test so that these assertion failures show up on OSSFuzz.